src/ape/api/transactions.py

```diff
@@ -231,10 +233,23 @@ def decode_logs(self, abi: Union[EventABI, "ContractEvent"]) -> Iterator[Contrac
         Returns:
             Iterator[:class:`~ape.types.ContractLog`]
         """
-        if not isinstance(abi, EventABI):
-            abi = abi.abi
+        if abi:
+            if not isinstance(abi, EventABI):
+                abi = abi.abi
 
-        yield from self.provider.network.ecosystem.decode_logs(abi, self.logs)
+            yield from self.provider.network.ecosystem.decode_logs(abi, self.logs)
+        else:
+            # if abi is not provided, decode all events
+            for log in self.logs:
+                contract_type = self.chain_manager.contracts.get(log["address"])
+                try:
+                    event_abi = contract_type.events[log["topics"][0]]  # type: ignore
```

Comment:
`contract_type` could be `None` here which would cause `AttributeError`
and if we fail to find a contract type, we should maybe raise an error

Reply:
Ah, I meant this line here: `contract_type = self.chain_manager.contracts.get(log["address"])`

`contract_type` is `Optional[ContractType]` so it may not have the `.events` attribute if it's `None`. I am surprised `mypy` is not picking it up though, so maybe I am missing something.

But in any case, I am happy to fix anything remaining and get this through :)